### PR TITLE
Add TPS command and improve the calculation of TPS

### PR DESCRIPTION
**Pull request by ReinforceZwei**

### CODE REVIEW

File: MinecraftClient/McClient.cs [hunk 1]
```diff
@@ -75,7 +75,14 @@ public class McClient : IMinecraftComHandler
...
         private DateTime lastTime;
-        private Double serverTPS = 0;
+        private double serverTPS = 0;
+        private double averageTPS = 20;
+        private const int maxSamples = 5;
+        private List<double> tpsSamples = new List<double>(maxSamples);
+        private double sampleSum = 0;
+
+        // players latency
+        private Dictionary<string, int> playersLatency = new Dictionary<string, int>();
```
Comment (ORelio) [inline MinecraftClient/McClient.cs:85]: I see this being populated but not used. What is its purpose?

File: MinecraftClient/McClient.cs [hunk 2]
```diff
@@ -1970,9 +1977,20 @@ public void OnTimeUpdate(long WorldAge, long TimeOfDay)
...
+                    {
+                        // full
+                        sampleSum -= tpsSamples[0];
+                        tpsSamples.RemoveAt(0);
+                    }
+                    tpsSamples.Add(tps);
+                    sampleSum += tps;
+                    averageTPS = sampleSum / tpsSamples.Count;
                     serverTPS = tps;
+                    //ConsoleIO.WriteLine("tps: " + tps + ", average tps: " + averageTPS);
```
Comment (ORelio) [inline MinecraftClient/McClient.cs:1993]: Seems like a forgotten debug test output.

File: MinecraftClient/McClient.cs (hunk 1 above)
Comment (ReinforceZwei) [reply MinecraftClient/McClient.cs:85]: I planned to make a command for viewing other player's ping. Also tried include player latency as tps calculate factor but seems like it doesn't affect much so yes, it is not used now.

**Comment by Nekiplay** [comment]
I made a team for myself a long time ago /tps
image

File: MinecraftClient/McClient.cs (hunk 1 above)
Comment (Nekiplay) [reply MinecraftClient/McClient.cs:85]: Perhaps my Chat Bot which I made a long time ago will help in creating this.
https://pastebin.com/ZCssAJih

**Comment by Nekiplay** [comment]
add else if?
image

**Comment by ORelio** [comment]
Seems good to me 👍

**Pull request merged**